fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/fluid/FluidVariant.java

```diff
@@ -43,6 +50,12 @@
 @ApiStatus.Experimental
 @ApiStatus.NonExtendable
 public interface FluidVariant extends TransferVariant<Fluid> {
+	public static final Codec<FluidVariant> CODEC = RecordCodecBuilder.create(instance ->
+			instance.group(
+				Registries.FLUID.getCodec().fieldOf("fluid").forGetter(FluidVariant::getFluid),
+				NbtCompound.CODEC.optionalFieldOf("nbt").forGetter(variant -> Optional.ofNullable(variant.getNbt()))
```

Comment:
If I remember correctly, the NBT parsing code might read small numbers as a different datatype than `int`, that should really be checked.